vcd/resource_vcd_vapp_access_control.go

```diff
@@ -0,0 +1,360 @@
+package vcd
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdAccessControlVapp() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAccessControlVappCreate,
+		Read:   resourceAccessControlVappRead,
+		Update: resourceAccessControlVappUpdate,
+		Delete: resourceAccessControlVappDelete,
+		Importer: &schema.ResourceImporter{
+			State: accessControlVappImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Required:    false,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"vapp_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "vApp identifier",
+			},
+			"shared_to_everyone": &schema.Schema{
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Whether the vApp is shared to everyone",
+			},
+			"everyone_access_level": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{types.ControlAccessReadOnly, types.ControlAccessReadWrite, types.ControlAccessFullControl}, true),
+				Description:  "Access level when the vApp is shared with everyone (one of ReadOnly, Change, FullControl). Required when shared_to_everyone is set",
+			},
+			"shared": &schema.Schema{
```

Comment:
OK, so this is probably me, but I find the use of the **past** tense `shared` across the PR's field names strange. Why not just simply `share`, `share_to_everyone`, etc. Like in UI:

<img width="431" alt="Screenshot 2020-08-25 at 16 52 03" src="https://user-images.githubusercontent.com/11330689/91182818-56726380-e6f3-11ea-87cd-dedc408879d8.png">

Maybe even clearer to name the block `sharing`?

Granted, I realize we had some other past tense fields in other resources:

```
# 2.10.0
cpu_hot_add_enabled = true
memory_hot_add_enabled = false
prevent_update_power_off = false

...
is_enabled        = "true"
```

Wonder what others think.

CC: @Didainius @vbauzysvmware

Reply:
changed to `shared_with`